File: Python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  silnia.py
#  
# Obliczanie silni liczby naturalnej
#  
#  

def silnia_it(n):
    """
    n! = 1 * 2 * ... * n
    """
    wynik = 1
    for i in range(1, n + 1):
        wynik = wynik * i
        print(wynik)
    
    return wynik
    


def main(args):
    n = 3 # wielokrotne przypisanie
    print("{}! silnia wynosi {}".
           format(n, silnia_it(n)))
           
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
 

File: cpp/silnia.cpp
/*
 * silnia.cpp
 * 
 */


#include <iostream>
using namespace std;

int silnia_it(int a){
    int wynik = 1;
    for (int i = 1; i <= a; i++){
        wynik = wynik * i;
    }
    return wynik;
}


int silnia_rek(int a){
    if (a < 2){
        return 1;
    }
    return a * silnia_rek(a - 1);
}

int main(int argc, char **argv)
{
	int a;
    cout << "Podaj liczbę: ";
    cin >> a;
    cout << "Silnia: " << silnia_it(a) << endl;
    cout << "Silnia: " << silnia_rek(a) << endl;
	return 0;
}

